internal/integrations/github/github.go

```diff
@@ -0,0 +1,47 @@
+package github
+
+import (
+	"context"
+	"errors"
+
+	"github.com/google/go-github/v26/github"
+	"golang.org/x/oauth2"
+)
+
+// GitHub handles integrations with the GitHub API
+type GitHub struct {
```

Comment:
bike-shedding: I've seen "github the corporation" stylized as GitHub, but "github the website/service" is typically stylized as Github. I'd prefer `Github` but don't feel too strongly either way, as long as it's consistent. Specifically, the config fields added in this PR are stylized as `Github*` which doesn't match this struct.